From af5f546a7a4dd634c8a3fc17a1688ad6f36bd521 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Wed, 27 Feb 2019 10:15:39 -0800
Subject: [PATCH 0515/1851] dt-bindings: display: xlnx: Add clock framework
 support to SDI Tx

commit d864740b14e141ba1973415479b2a7d4b88e8d39 from
https://github.com/Xilinx/linux-xlnx.git

This patch adds support for clock framework to SDI Tx driver by adding
and making clocks and clock-names as required properties.

This breaks backwards compatibility as the IP version hasn't changed.
So compatible string isn't changed. But new required clock properties
are added.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../devicetree/bindings/display/xlnx/xlnx,sdi-tx.txt         | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Documentation/devicetree/bindings/display/xlnx/xlnx,sdi-tx.txt b/Documentation/devicetree/bindings/display/xlnx/xlnx,sdi-tx.txt
index 093d53321e0f..971ac5304761 100644
--- a/Documentation/devicetree/bindings/display/xlnx/xlnx,sdi-tx.txt
+++ b/Documentation/devicetree/bindings/display/xlnx/xlnx,sdi-tx.txt
@@ -13,6 +13,9 @@ Required properties:
    Documentation/devicetree/bindings/graph.txt.
    Minimum one port is required. At max, 2 ports are present.
    The reg index for AXI4 stream port is 0 and for ancillary data is 1.
+ - clocks: List of phandles to AXI Lite, Video and SDI Tx Clock
+ - clock-names: Must contain "s_axi_aclk", "video_in_clk" and "sdi_tx_clk"
+   in same order as clocks listed in clocks property.
 
 Optional properties:
  - xlnx,vpss: vpss phandle
@@ -29,6 +32,8 @@ Example:
 		#address-cells = <1>;
 		#size-cells = <0>;
 		xlnx,vpss = <&v_proc_ss_0>;
+		clock-names = "s_axi_aclk", "video_in_clk", "sdi_tx_clk";
+		clocks = <&misc_clk_0>, <&misc_clk_1>, <&misc_clk_2>;
 		ports {
 			#address-cells = <1>;
 			#size-cells = <0>;
-- 
2.31.1

